### PR TITLE
Advanced Front-end - Lesson1 obligatory exercise and Lesson2 Git exercise.

### PR DESCRIPTION
**Pull request by AndreyWeber**
1. Project environment setup: npm config, bower configs.
2. Git setup: gitignore, gitconfig.
3. App server based on node.js express package, implemented in app.js file. Application can be started with "npm start" command.
4. Implemented Lesson1 obligatory exercise: Handlebars helper for JSON formatted output {{json jsonObj}}; Handlebars block helper for striped table like data output {{#stripedTable oddRowClass evenRowClass}}.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "node_modules",
+    "bower_components",
+    "test",
+    "tests"
+  ],
+  "dependencies": {
+    "underscore": "1.8.3",
+    "backbone": "^1.0.0",
+    "react": ">0.1.0",
+    "angular": "^1.5.5",
```
Comment (inaccessible) [inline bower.json]: > any version

please check documentation for "any version"
https://github.com/npm/node-semver

File: gitconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[user]
```
Comment (inaccessible) [inline gitconfig]: it's your busies but not sure that it's good idea to have `giticonfig` int repo

File: client_src/test.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return new Handlebars.SafeString('<pre>' + jsonString + '</pre>');
+    }
+
+    function renderStripedTableHelper (context, options) {
+        if (!context) {
+            return new Handlebars.SafeString('<div class="striped-table__odd-row"><i>&lt;Empty striped table&gt;</i></div>')
+        }
+
+        var result = context.map(function (item, idx) {
+            var rowClass = (idx + 1) % 2
```
Comment (inaccessible) [inline client_src/test.js]: `idx + 1` why + 1?

**Comment by AndreyWeber** [comment]
All notes are taken into account and fixed.

File: bower.json (hunk 1 above)
Comment (AndreyWeber) [inline bower.json]: Fixed

File: gitconfig (hunk 2 above)
Comment (AndreyWeber) [inline gitconfig]: Fixed. Config removed.

File: client_src/test.js (hunk 3 above)
Comment (AndreyWeber) [inline client_src/test.js]: Fixed as discussed.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+**/.idea/
```
Comment (inaccessible) [inline .gitignore]: do you really need `**` ?

**Comment by AndreyWeber** [comment]
*\* wildcard removed for .idea folder, it is not needed, indeed.

**Pull request merged**